polyfill.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const ES = require('es-abstract');
+
+if (typeof Symbol !== 'function' || !Symbol.isConcatSpreadable) {
+  throw new Error('Symbol.isConcatSpreadable missing from the environment');
+}
+
+if (typeof Array.prototype.flatMap !== 'function') {
+  // eslint-disable-next-line no-extend-native
+  Object.defineProperty(Array.prototype, 'flatMap', {
+    configurable: true,
+    writable: true,
+    // https://tc39.github.io/proposal-flatMap/#sec-Array.prototype.flatMap
+    value: function flatMap(callback, thisArg = undefined) {
+      const O = ES.ToObject(this);
+      const A = new (ES.SpeciesConstructor(O, Array))(0);
+      flattenIntoArray(A, O, 0, 1, callback, thisArg);
```

Comment:
the spec has changed since this initial PR; you may need to do some updates here. In particular, `O.length` is passed in as an argument to `FlattenIntoArray`